### PR TITLE
Projectiles

### PR DESCRIPTION
**Pull request by Steampunkery**
# Description
This PR modifies the engine to allow projectiles to be added from any module through the use of a `projectilesConfig.json` file in the module's `assets/configs` folder.

# Testing
Test it by adding a custom `projectilesConfig.json`

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/projectile/ProjectileConfigs.java [hunk 1]
```diff
@@ -38,40 +41,46 @@
     public ProjectileConfigs(OggSoundManager soundManager, EffectTypes effectTypes, GameColors cols) {
         myConfigs = new HashMap<>();
 
-        Json json = Assets.getJson("core:projectilesConfig");
-        JsonValue rootNode = json.getJsonValue();
+        Set<ResourceUrn> projectileConfigurationFiles = Assets.getAssetHelper().list(Json.class,
+                "[a-z]*:projectilesConfig");
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/projectile/ProjectileConfigs.java]: There are way longer lines than this in the PR. Either break them all into parts, or don't break this one.

File: engine/src/main/java/org/destinationsol/game/projectile/ProjectileConfigs.java [hunk 2]
```diff
@@ -38,40 +41,46 @@
...
+                boolean zeroAbsSpd = node.getBoolean("zeroAbsSpd", false);
+                Vector2 origin = SolMath.readV2(node.getString("texOrig", "0 0"));
+                float acc = node.getFloat("acceleration", 0);
+                String workSoundUrn = node.getString("workSound", "");
+                OggSound workSound = workSoundUrn.isEmpty() ? null : soundManager.getSound(workSoundUrn);
+                boolean bodyless = node.getBoolean("massless", false);
+                float density = node.getFloat("density", -1);
+                float dmg = node.getFloat("dmg");
+                float emTime = node.getFloat("emTime", 0);
+                ProjectileConfig c = new ProjectileConfig(tex, texSz, spdLen, stretch, physSize, dmgType,
```
Comment (vampcat) [inline engine/src/main/java/org/destinationsol/game/projectile/ProjectileConfigs.java]: Can we rename c to something more.. meaningful? `projectileConfig` would be ideal.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**